### DISCUSSION TITLE
Tab text vs. window text

### DISCUSSION THREAD

**Discussion by vefatica** [microsoft/terminal#18273] [General]
What's going on?

![Image](https://github.com/user-attachments/assets/b248531b-c8c3-4f97-96d0-8765fb81c466)

![Image](https://github.com/user-attachments/assets/3794e99f-3a5b-4ebd-a5e7-3fb0970d3b53)

**Comment by DHowett** [comment]
Sorry, can you explain the problem you're seeing?

There are hidden windows that have titles. I'm assuming you tore 12144 out of a Terminal window, so we hid it and never changed the title.

The tab is not a window, and therefore none of the tabs will show up in Spy++.

**Comment by vefatica** [comment]
What about the hidden windows?  Can't their titles be accurate?  The problem is that what I see doesn't seem to correspond to reality.

**Reply by DHowett** [reply]
Why bother? They don't contain anything.

**Comment by vefatica** [comment]
I'm trying to figure out what window my shell is in, that's all.  Tearout/in has made doing that more complicated.

**Reply by DHowett** [reply]
Tearout/merging only made it more complicated if your programming model was built with the incorrect assumption that console applications could never be hosted anywhere but a single window. 🙂 

Using the title to detect which window holds your current shell is never going to be reliable. Terminal might one day update to put tab information (like "Command Prompt (+3 more)") in the title, or worse. You could disable title updates. You could be running one shell in a non-active pane, and its title will *never* make it out to the window until it becomes focused.

Use `GetAncestor(GetConsoleWindow(), GA_ROOTOWNER)` and you will always get the window hosting your shell. Anything else is just a poor heuristic.

**Comment by vefatica** [comment]
I don't use the title.  My notion of the Cascadia window is GetParent(GetConsoleWindow()).

Is that reliable?  Will GetAncestor() with GA_ROOTOWNER work any better?

I reckon one of those will continue to work if I can determine (without polling) that I need to check again.  I'm thinking of a WinEvent hook (EVENT_OBJECT_CREATE/DESTROY) but I haven't tried it yet.

**Reply by lhecker** [reply]
`GetAncestor(GetConsoleWindow(), GA_ROOTOWNER)`, the way Dustin suggested it, is the correct choice for getting the "parent-most" window. `GetParent` is not robust enough for that purpose.

**Comment by OS-Scratch** [comment]
If you want to know your window title in the UI, you should disable Show title bar (or something similar) in Settings.
